addons/docs/src/frameworks/vue/sourceDecorator.ts

```diff
@@ -30,47 +30,45 @@ export const sourceDecorator = (storyFn: any, context: StoryContext) => {
     return story;
   }
 
-  try {
-    // Creating a Vue instance each time is very costly. But we need to do it
-    // in order to access VNode, otherwise vm.$vnode will be undefined.
-    // Also, I couldn't see any notable difference from the implementation with
-    // per-story-cache.
-    // But if there is a more performant way, we should replace it with that ASAP.
-    const vm = new Vue({
-      data() {
-        return {
-          STORYBOOK_VALUES: context.args,
-        };
-      },
-      render(h) {
-        return h(story);
-      },
-    }).$mount();
-
-    const channel = addons.getChannel();
-
-    const storyComponent = getStoryComponent(story.options.STORYBOOK_WRAPS);
-
-    const storyNode = lookupStoryInstance(vm, storyComponent);
-
-    const code = vnodeToString(storyNode._vnode);
-
-    channel.emit(
-      SNIPPET_RENDERED,
-      (context || {}).id,
-      prettier.format(`<template>${code}</template>`, {
-        parser: 'vue',
-        plugins: [prettierHtml],
-        // Because the parsed vnode missing spaces right before/after the surround tag,
-        // we always get weird wrapped code without this option.
-        htmlWhitespaceSensitivity: 'ignore',
-      })
-    );
-  } catch (e) {
-    logger.warn(`Failed to generate dynamic story source: ${e}`);
-  }
+  const channel = addons.getChannel();
+
+  const storyComponent = getStoryComponent(story.options.STORYBOOK_WRAPS);
 
-  return story;
+  return {
+    components: {
+      Story: story,
+    },
+    // We need to wait until the wrapper component to be mounted so Vue runtime
+    // struct VNode tree. We get `this._vnode == null` if switch to `created`
+    // lifecycle hook.
+    mounted() {
```

Comment:
It seems every control change triggers re-mount so no need to use `updated` hook?

Reply:
Yeah, I believe that is the current implementation.